src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClientExceptions.cs

```diff
@@ -22,6 +23,12 @@ public class ServerNotAvailableException : DiagnosticsClientException
         public ServerNotAvailableException(string msg) : base(msg) {}
     }
 
+    // When process does not have authorization to enumerate NamedPipes.
+    public class NamedPipeEnumerationUnauthorizedException : DiagnosticsClientException
```

Comment:
That's my bad - my comment was confusing. Given that we don't do any special handling of this exception - no need to create a new one. We can just keep using `DiagnosticsClientException`, just with the new constructor you added.

Reply:
Do we want to make the reason behind the exception being thrown clearer and add a more detailed error message?